From: Dan Carpenter <dan.carpenter@oracle.com>
Date: Thu, 25 Jul 2013 19:48:32 +0300
Subject: [PATCH 051/223] RDMA/cxgb4: Fix stack info leak in c4iw_create_qp()
Patch-mainline: v3.11-rc4
Git-commit: ae1fe07f3f422c62a0baf6a777141d7110e8a111
References: bsc#909577 FATE#317550

"uresp.ma_sync_key" doesn't get set on this path so we leak 8 bytes of data.

Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
Acked-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
Acked-by: David Chang <dchang@suse.com>
Signed-off-by: John Jolly <jjolly@suse.de>
---
 drivers/infiniband/hw/cxgb4/qp.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/drivers/infiniband/hw/cxgb4/qp.c
+++ b/drivers/infiniband/hw/cxgb4/qp.c
@@ -1663,6 +1663,8 @@ struct ib_qp *c4iw_create_qp(struct ib_p
 		if (mm5) {
 			uresp.ma_sync_key = ucontext->key;
 			ucontext->key += PAGE_SIZE;
+		} else {
+			uresp.ma_sync_key =  0;
 		}
 		uresp.sq_key = ucontext->key;
 		ucontext->key += PAGE_SIZE;
